File: CPP/06_Noelia_Banios-AlineacionesCopa_Programacion/alineacionesCopa1.cpp
#include <iostream>
#include <windows.h>
#include <ctime>
/*****************************/
/*  HECHO POR @NOELIA BAÑOS  */
/*****************************/
using namespace std;
/******************************/
/*         CONSTANTES         */
/******************************/
const int POSICIONES = 5;
/******************************/
/*  DECLARACION DE FUNCIONES  */
/******************************/
void meterDatos(int array1[], int array2[], int POSICIONES);
/*********************/
/*        MAIN       */
/*********************/
int main()
{
    SetConsoleOutputCP(CP_UTF8); // Consola en español
    srand(time(NULL));
    // VARIABLES
    int array1[POSICIONES];
    int array2[POSICIONES];
    // RELLENADO DE DATOS DEL PRIMER ARRAY
    for (int i = 0; i < POSICIONES; i++)
    {
        array1[i] = rand() % 20;
    }
    meterDatos(array1, array2, POSICIONES);

    // SIGUIENTE
    cout << "\n------------------------" << endl;
    int elecc, num;

    // BUCLE EN EL QUE PREGUNTAMOS LA POSICIÓN DEL NUMERO QUE SE VA A CAMBIAR
    do
    {
        cout << "¿En qué posición quieres introducir el número?" << endl;
        cin >> elecc;
        cout << "¿Qué número quieres introducir?" << endl;
        cin >> num;

        array1[elecc - 1] = num;
        meterDatos(array1, array2, POSICIONES);

    } while (num == 0);

    return 0;
}
/*************************************************************************/
/*                          FUNCION METER DATOS                          */
/*************************************************************************/
/* Añadimos los datos que necesitamos para que entre los array           */
/* se cambien las posiciones que necesitamos                             */
/*        Entrada: ambos arrays y las posiciones que se usan en el array */
/*        Salida: nunguna                                                */
/*************************************************************************/
void meterDatos(int array1[], int array2[], int POSICIONES)
{
    // PASO DE DATOS AL OTRO ARRAY
    for (int i = 0; i < POSICIONES; i++)
    {
        array2[i] = array1[4 - i];
    }

    // MOSTRAR POR PANTALLA EL PRIMERO
    cout << "Primer array" << endl;
    for (int i = 0; i < POSICIONES; i++)
    {
        cout << array1[i] << " | ";
    }

    // MOSTRAR POR PANTALLA EL SEGUNDO
    cout << "\nSegundo array" << endl;
    for (int i = 0; i < POSICIONES; i++)
    {
        cout << array2[i] << " | ";
    }
    cout << endl;
}

File: JAVA/tareaClases_0/alineacionesCopa1.java
package tareaClases_0;

import java.util.Scanner;

/**************************/
/* HECHO POR @NOELIA BAÑOS*/
/**************************/

public class alineacionesCopa1 {

	public static void main(String[] args) {
		Scanner teclado = new Scanner(System.in);
		// VARIABLES
		int posiciones = 5;
		int array1[] = new int[posiciones];
		int array2[] = new int[posiciones];

		// RELLENADO DE DATOS DEL PRIMER ARRAY
		for (int i = 0; i < posiciones; i++) {
			array1[i] = (int) (Math.random() * 20);
		}
		meterDatos(array1,array2, posiciones);
		

		// SIGUIENTE
		System.out.println("\n------------------------");
		int elecc, num;
// BUCLE EN EL QUE PREGUNTAMOS LA POSICIÓN DEL NUMERO QUE SE VA A CAMBIAR
		do {
			System.out.println("¿En que posición quieres introducir el número?");
			elecc = teclado.nextInt();
			System.out.println("¿Qué número quieres introducir?");
			num = teclado.nextInt();

			array1[elecc - 1] = num;
			meterDatos(array1,array2, posiciones);

		} while (num == 0);
		 teclado.close();
	}
	
	// FUNCION QUE USAMOS DOS VECES Y POR ESO LA PONEMOS FUERA
	public static void meterDatos(int array1[],int array2[], int posiciones){
		// PASO DE DATOS AL OTRO ARRAY
				for (int i = 0; i < posiciones; i++) {
					array2[i] = array1[4 - i];
				}

				// MOSTRAR POR PANTALLA EL PRIMERO
				System.out.println("Primer array");
				for (int i = 0; i < posiciones; i++) {
					System.out.print(array1[i] + " | ");
				}

				// MOSTRAR POR PANTALLA EL SEGUNDO
				System.out.println("\nSegundo array");
				for (int i = 0; i < posiciones; i++) {
					System.out.print(array2[i] + " | ");
				}
	}
}
